src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -100,10 +100,9 @@ class DefaultDataContainer implements DataContainer {
     await cache.cache(categories, context)
   }
 
-  setCities = (cities: Array<CityModel>) => {
-    // TODO: Offline available cities will be persisted in NATIVE-175. For now switching cities when offline is not possible.
+  setCities = async (cities: Array<CityModel>) => {
     this._cities = cities
-    return Promise.resolve()
+    await this._databaseConnector.storeCities(cities)
```

Comment:
I'm not 100% sure I think you have to return the promise. Else it is just resolved as soon as the function returns.
```suggestion
    return this._databaseConnector.storeCities(cities)
```